File: lib-util/src/main/java/code/util/MD5.java
package code.util;

import java.io.File;
import java.io.FileInputStream;
import java.security.MessageDigest;
import java.security.NoSuchAlgorithmException;

public class MD5 {

    public static String encrypt(String string) {
        if (string == null || string.length() == 0) {
            return "";
        }
        return encrypt(string.getBytes());
    }

    public static String encrypt(byte[] bytes) {
        if (null == bytes || bytes.length == 0) {
            return "";
        }
        try {
            MessageDigest md5 = MessageDigest.getInstance("MD5");
            byte[] md5Bytes = md5.digest(bytes);
            StringBuilder result = new StringBuilder();
            for (byte b : md5Bytes) {
                String temp = Integer.toHexString(b & 0xff);
                if (temp.length() == 1) {
                    temp = "0" + temp;
                }
                result.append(temp);
            }
            return result.toString();
        } catch (NoSuchAlgorithmException e) {
            e.printStackTrace();
        }
        return "";
    }

    // 计算文件的 MD5 值
    public static String encrypt(File file) {
        if (file == null || !file.isFile() || !file.exists()) {
            return "";
        }
        FileInputStream in = null;
        StringBuilder result = new StringBuilder();
        byte buffer[] = new byte[8192];
        int len;
        try {
            MessageDigest md5 = MessageDigest.getInstance("MD5");
            in = new FileInputStream(file);
            while ((len = in.read(buffer)) != -1) {
                md5.update(buffer, 0, len);
            }
            byte[] bytes = md5.digest();

            for (byte b : bytes) {
                String temp = Integer.toHexString(b & 0xff);
                if (temp.length() == 1) {
                    temp = "0" + temp;
                }
                result.append(temp);
            }
        } catch (Exception e) {
            e.printStackTrace();
        } finally {
            IOUtil.close(in);
        }
        return result.toString();
    }

}

File: lib-util/src/main/java/code/util/MD5.kt
package code.util

import code.util.IOUtil.close
import java.io.File
import java.io.FileInputStream
import java.security.MessageDigest
import java.security.NoSuchAlgorithmException

object MD5 {
    
    fun encrypt(string: String?): String {
        return if (string.isNullOrEmpty()) {
            ""
        } else encrypt(string.toByteArray())
    }

    fun encrypt(bytes: ByteArray?): String {
        if (null == bytes || bytes.isEmpty()) {
            return ""
        }
        try {
            val md5 = MessageDigest.getInstance("MD5")
            val md5Bytes = md5.digest(bytes)
            val result = StringBuilder()
            for (b in md5Bytes) {
                var temp = Integer.toHexString(b.toInt() and 0xff)
                if (temp.length == 1) {
                    temp = "0$temp"
                }
                result.append(temp)
            }
            return result.toString()
        } catch (e: NoSuchAlgorithmException) {
            e.printStackTrace()
        }
        return ""
    }

    // 计算文件的 MD5 值
    fun encrypt(file: File?): String {
        if (file == null || !file.isFile || !file.exists()) {
            return ""
        }
        var input: FileInputStream? = null
        val result = StringBuilder()
        val buffer = ByteArray(8192)
        var len: Int
        try {
            val md5 = MessageDigest.getInstance("MD5")
            input = FileInputStream(file)
            while (input.read(buffer).also { len = it } != -1) {
                md5.update(buffer, 0, len)
            }
            val bytes = md5.digest()
            for (b in bytes) {
                var temp = Integer.toHexString(b.toInt() and 0xff)
                if (temp.length == 1) {
                    temp = "0$temp"
                }
                result.append(temp)
            }
        } catch (e: Exception) {
            e.printStackTrace()
        } finally {
            close(input)
        }
        return result.toString()
    }
}